### PR TITLE
blocks: reserve memory in vector_sink constructor

### PR DESCRIPTION
**Pull request by noc0lour**
Might bring some performance benefits. Changes public interface to vector_sink.

### CODE REVIEW

**Comment by willcode** [comment]
How do we know whether blocks like vector_sink are subclassed by OOT projects? I'm asking mainly because I started working on file_source offset/len and was wondering the same thing for that block?

**Comment by noc0lour** [comment]
We don't know and if a certain OOT uses a block. Though in this case by using a default constructor this shouldn't be the case. (Most certainly a lot of OOTs are using a vector_sink for QA).

**Comment by noc0lour** [comment]
Adds features proposed in #1548

**Comment by marcusmueller** [comment]
Hey, @noc0lour, so, this is a nice, short, clear-benefit PR, but:

This would do a lot better with a sensible default (maybe a page size. Who needs 100 samples?), and honestly: I think you should combine with the proposed *max* feature, as saving a gr::blocks::head definitely is as much a perf improvement.

**Comment by noc0lour** [comment]
Will do that, The value is super application specific, so taking `100` was just merely a PoC than a `real` number for inclusion.

**Comment by noc0lour** [comment]
Was looking at the `get_page` function in runtime, unfortunately this function is not in the public headers and I object to copying it or cross linking through the tree.

**Comment by marcusmueller** [comment]
Tbh, I'll be content with you having `4096 / sizeof(element)` in the code and a comment that says `//4kB is a good assumption on page sizes`.

Other than that, if anything, `gnuradio-runtime` is the component that is OK to be cross-used all over the tree... but I agree, let's not unnecessarily expose inner API. However, that `gr::getpagesize` thing was definitely *meant* to be externally accessible, so I'm not that critical there. The `fprintf(err` would have to go, though...

**Comment by mbr0wn** [comment]
Even a value like 1024 would be fine. I would suggest *not* combining this with a max feature, but rather make that a separate PR.

**Comment by marcusmueller** [comment]
I thought about this. The name "minimum items" suggest something breaks when the user doesn't supply at least as many – and that's really not the case. Instead, I'd recommend a "Items of Memory Reservation" as GRC property name, and "reserve_items" as parameter name. With these wording corrections, I'd be willing to give this a rather immediate merge – there's nothing to be lost here, I guess.

**Comment by noc0lour** [comment]
Added 1024 items and rename min_items -> reserve_items

File: gr-blocks/grc/blocks_vector_sink_x.xml [hunk 1]
```diff
@@ -45,6 +45,12 @@
 		<value>1</value>
 		<type>int</type>
 	</param>
+	<param>
+		<name>Minimum data items</name>
```
Comment (marcusmueller) [inline gr-blocks/grc/blocks_vector_sink_x.xml]: Should read `Reserve Memory for items` or something

File: gr-blocks/grc/blocks_vector_sink_x.xml [hunk 2]
```diff
@@ -45,6 +45,12 @@
...
+		<key>reserve_items</key>
+		<value>100</value>
```
Comment (marcusmueller) [inline gr-blocks/grc/blocks_vector_sink_x.xml]: 1024

**Comment by marcusmueller** [comment]
Ah! I'm taking the freedom of updating the copyright year myself while merging, @noc0lour.

**Comment by noc0lour** [comment]
sure, go ahead

**Comment by marcusmueller** [comment]
Manually merged (b52726077244d2b8439c663144e5f013fa3d8557). Closing this PR! Thanks for the great work 👍

**Pull request closed without merging**